es-ds-components/components/es-popover.vue

```diff
@@ -0,0 +1,105 @@
+<script setup lang="ts">
+import OverlayPanel from 'primevue/overlaypanel';
+
+const props = defineProps({
+    target: {
+        type: String,
+        required: true,
+    },
+    triggers: {
+        type: String,
+        required: false,
+        default: 'focus',
+    },
+    show: {
+        type: Boolean,
+        required: false,
+        default: false,
+    },
+    variant: {
+        type: String,
+        required: false,
+        default: 'dark',
+        validator: (val: string) => ['light', 'dark'].includes(val as string),
+    },
+});
+
+const op = ref<InstanceType<typeof OverlayPanel> | null>(null);
+const slots = useSlots();
+const hasTitle = computed(() => !!slots.title);
+const closePanel = () => {
+    op.value?.hide();
+};
+const showPanel = (event: Event) => {
+    op.value?.show(event);
+};
+
+onMounted(() => {
+    const targetElement = document.getElementById(props.target);
+    if (targetElement) {
+        const triggers = props.triggers.split(' ').map(trigger => trigger === 'hover' ? 'mouseover' : trigger);
+        for (const trigger of triggers) {
+            targetElement.addEventListener(trigger, showPanel);
+        }
+    }
+});
+
+watch(
+    () => props.show,
+    (newVal) => {
+        if (newVal) {
+            const targetElement = document.getElementById(props.target);
```

Comment:
🤔 should this be wrapped in an `if(import.meta.client)`?

Reply:
The onMounted hook is [not called during server side rendering](https://vuejs.org/api/composition-api-lifecycle#onmounted), so I don't think it matters

Reply:
> This hook is typically used for performing side effects that need access to the component's rendered DOM, or for limiting DOM-related code to the client in a [server-rendered application](https://vuejs.org/guide/scaling-up/ssr).